ui/components/app/snaps/update-snap-permission-list/update-snap-permission-list.js

```diff
@@ -37,75 +46,103 @@ export default function UpdateSnapPermissionList({
   );
 
   const snapsMetadata = useSelector(getSnapsMetadata);
-  const snapsNameGetter = getSnapName(snapsMetadata);
 
-  const approvedPermissionsToShow = {
+  const approvedCombinedPermissions = {
     ...approvedPermissions,
     connection_permission: approvedConnections ?? {},
   };
 
-  const revokedPermissionsToShow = {
+  const revokedCombinedPermissions = {
     ...revokedPermissions,
     connection_permission: revokedConnections ?? {},
   };
 
-  const newPermissionsToShow = {
+  const newCombinedPermissions = {
     ...newPermissions,
     connection_permission: newConnections ?? {},
   };
 
+  const [showAll, setShowAll] = useState(false);
+
+  const newWeightedPermissions = getWeightedPermissions({
```

Comment:
Do we not want to filter new or revoked permissions?

Reply:
[The following statement is the requirement](https://github.com/MetaMask/snaps/issues/2235#issuecomment-2025221206):
```Always show requested and revoked permissions, and approved permissions with weight <= 2```
So no filtering for requested and revoked on update, as far as I understood. Only _less important_ approved are hidden.

Reply:
😅 - sorry this is getting complicated, I once again had a different understanding. I thought we wanted to start filtering revoked and approved permissions since they are less important that the newly requested permissions.

Perhaps we need to have a chat Erik, you and I once everyone is back.. 😅

Reply:
This is also discussed on the meeting. So, the decision was to keep logic the same, just fill the gap if needed with adding approved permissions if there is not enough permissions to show.